src/core/core/managers/queue_manager.py

```
@@ -1075,36 +1086,82 @@ def post_collection_bots(self, source_id: str, user_id: str | None = None):
+            user_id=user_id,
+            trigger_dependents=False,
+        ):
+            return {"error": "Could not schedule dependent bot"}, 500
+        return {"message": "Dependent bots scheduled"}, 200
+
+    def _enqueue_bot_graph(
+        self,
+        bots: Sequence[Any],
+        dependencies_by_id: dict[str, list[str]],
+        *,
+        filter: dict | None,
```

Ruff (0.15.20)

[error] Function argument `filter` is shadowing a Python builtin

(A002)